src/types.js

```diff
@@ -234,8 +234,14 @@ export type MessageLike =
       ...Outbox,
     }>;
 
-// From docs: https://formatjs.io/docs/react-intl/api/#formatmessage
-type IntlMessageFormatValue = string | number | boolean | null | void;
+// We adapt `IntlMessageFormatValue` from the `react-intl` libdef,
+// which doesn't quite match the doc at
+// https://formatjs.io/docs/react-intl/api/#formatmessage.
+type PrimitiveType = string | number | boolean | null | void | Date;
+// (name is from libdef)
+// eslint-disable-next-line flowtype/type-id-match
+type FormatXMLElementFn = (parts: string[]) => string;
+type IntlMessageFormatValue = PrimitiveType | FormatXMLElementFn;
```

Comment:
Ah, but also I think this can all be avoided by correcting the libdef like so -- this is something that TS doesn't know how to express, so flowgen wouldn't have been able to translate it:
```diff
     formatMessage(
       descriptor: MessageDescriptor,
-      values?: { [key: string]: PrimitiveType | FormatXMLElementFn<string, string>, ... },
+      values?: { +[key: string]: PrimitiveType | FormatXMLElementFn<string, string>, ... },
     ): string;
```
The `+` makes the property (here, the "indexer property" so effectively all properties) "covariant" rather than "invariant" for the object type. I.e., it's fine to pass a `values` where all the values have type some *subtype* of the complicated type mentioned, not only where they all have that type. Flow's docs don't quite explain this in one coherent place, but these seem to be the relevant bits:
https://flow.org/en/docs/types/interfaces/#toc-interface-property-variance-read-only-and-write-only
https://flow.org/en/docs/types/utilities/#toc-readonly
https://flow.org/en/docs/lang/variance/

and on the general concept independent of Flow, this is a pretty good explanation:
https://www.stephanboyer.com/post/132/what-are-covariance-and-contravariance

The case where that covariant type would be wrong is if `formatMessage` were going to mutate the object, and then we were going to look back at it and look at the new values it put there. Then it'd be important that the caller have the exact same idea of what types are allowed for those values as `formatMessage` itself does. But it shouldn't be mutating the object, so covariance is appropriate.